src/main/java/frc/robot/subsystems/Intake.java

```diff
@@ -186,6 +193,11 @@ public void simulationPeriodic() {
 
     // Update the Mechanism Arm angle based on the simulated arm angle
     m_arm.setAngle(Units.radiansToDegrees(m_armSim.getAngleRads()));
+
+    if (isAtUpPosition()) {
+      m_encoder.setPosition(Constants.IntakeConstants.kArmUpPosition);
```

Comment:
Which means this doesn't actually set the position for the contoller loop. Have the limit switch set the voltage on the motor to 0, so that stops the motor as instantly as possible.